internal/command/user_v2_model.go

```diff
@@ -0,0 +1,491 @@
+package command
+
+import (
+	"context"
+	"time"
+
+	"golang.org/x/text/language"
+
+	"github.com/zitadel/zitadel/internal/crypto"
+	"github.com/zitadel/zitadel/internal/domain"
+	"github.com/zitadel/zitadel/internal/eventstore"
+	"github.com/zitadel/zitadel/internal/repository/user"
+)
+
+type UserV2WriteModel struct {
+	eventstore.WriteModel
+
+	UserName string
+
+	MachineWriteModel bool
+	Name              string
+	Description       string
+	AccessTokenType   domain.OIDCTokenType
+
+	MachineSecretWriteModel bool
+	ClientSecret            *crypto.CryptoValue
+
+	ProfileWriteModel bool
+	FirstName         string
+	LastName          string
+	NickName          string
+	DisplayName       string
+	PreferredLanguage language.Tag
+	Gender            domain.Gender
+
+	AvatarWriteModel bool
+	Avatar           string
+
+	HumanWriteModel      bool
+	InitCode             *crypto.CryptoValue
+	InitCodeCreationDate time.Time
+	InitCodeExpiry       time.Duration
+	InitCheckFailedCount uint64
+
+	PasswordWriteModel       bool
+	PasswordEncodedHash      string
+	PasswordChangeRequired   bool
+	PasswordCode             *crypto.CryptoValue
+	PasswordCodeCreationDate time.Time
+	PasswordCodeExpiry       time.Duration
+	PasswordCheckFailedCount uint64
+
+	EmailWriteModel       bool
+	Email                 domain.EmailAddress
+	IsEmailVerified       bool
+	EmailCode             *crypto.CryptoValue
+	EmailCodeCreationDate time.Time
+	EmailCodeExpiry       time.Duration
+	EmailCheckFailedCount uint64
+
+	PhoneWriteModel       bool
+	Phone                 domain.PhoneNumber
+	IsPhoneVerified       bool
+	PhoneCode             *crypto.CryptoValue
+	PhoneCodeCreationDate time.Time
+	PhoneCodeExpiry       time.Duration
+	PhoneCheckFailedCount uint64
+
+	StateWriteModel bool
+	UserState       domain.UserState
+}
+
+func NewUserExistsWriteModel(userID, resourceOwner string) *UserV2WriteModel {
+	return newUserV2WriteModel(userID, resourceOwner, true, true, false, false, false, false, false, false)
+}
+
+func NewUserStateWriteModel(userID, resourceOwner string) *UserV2WriteModel {
+	return newUserV2WriteModel(userID, resourceOwner, true, true, false, false, false, false, true, false)
+}
+
+func NewUserHumanWriteModel(userID, resourceOwner string, profileWM, emailWM, phoneWM, passwordWM, avatarWM bool) *UserV2WriteModel {
+	return newUserV2WriteModel(userID, resourceOwner, true, false, profileWM, emailWM, phoneWM, passwordWM, true, avatarWM)
+}
+
+func newUserV2WriteModel(userID, resourceOwner string, humanWM, machineWM, profileWM, emailWM, phoneWM, passwordWM, stateWM, avatarWM bool) *UserV2WriteModel {
```

Comment:
just as a note: we should check if there's any better solution for future cases
a list of booleans is hard to read and track, e.g. if i'm not mistaken, the avatarWM is never set to true...

Reply:
Tried to separate it into functions, so that we could add it as we do with providers with the options.

Reply:
was more meant as a side node, but actually the idea is cool 😃